packages/block-editor/src/hooks/color.js

```diff
@@ -383,12 +384,27 @@ function useBlockProps( {
 		)?.color;
 	}
 
-	return addSaveProps( { style: extraStyles }, name, {
+	const saveProps = addSaveProps( { style: extraStyles }, name, {
 		textColor,
 		backgroundColor,
 		gradient,
 		style,
 	} );
+
+	const hasBackgroundValue =
+		backgroundColor ||
+		style?.color?.background ||
+		gradient ||
+		style?.color?.gradient;
+
+	return {
+		...saveProps,
+		className: classnames(
+			saveProps.className,
+			// Add background image classes in the editor, if not already handled by background color values.
+			! hasBackgroundValue && getBackgroundImageClasses( style )
```

Comment:
Thanks for confirming!

> Not sure, but maybe worth it one day 😄

It very well could be! I think it'd likely be handy to consolidate classname generation at some point, if only so that it's easier to debug / introspect where and how a block gets a particular class. It'd also be nice if it was clearer which classes receive classnames that are serialized and which ones are dynamically generated.

That said, I don't think it's particularly urgent, but it's a good thing to keep in the back of our minds 🙂